lib/tests/test_merged_tree.rs

```diff
@@ -1182,3 +1183,153 @@ fn test_merge_simplify_result() {
     let merged = side1_merged.merge(&base1_merged, &side2_merged).unwrap();
     assert_eq!(merged, expected_merged);
 }
+
+/// Test that we simplify content-level conflicts before passing them to
+/// files::merge().
+///
+/// This is what happens when you squash a conflict resolution into a conflict
+/// and it gets propagated to a child where the conflict is different.
+#[test]
+fn test_merge_simplify_file_conflict() {
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    let conflict_path = RepoPath::from_internal_string("CHANGELOG.md");
+    let other_path = RepoPath::from_internal_string("other");
+
+    let text1 = r#"### New features
+
+* The `ancestors()` revset function now takes an optional `depth` argument
+  to limit the depth of the ancestor set. For example, use `jj log -r
+  'ancestors(@, 5)` to view the last 5 commits.
+
+* Support for the Watchman filesystem monitor is now bundled by default. Set
+  `core.fsmonitor = "watchman"` in your repo to enable.
+
+### Fixed bugs 
+"#;
+    let text2 = r#"### New features
```

Comment:
After the naming below is finalized, it'd be nice to rename this from `text2` to something like `text_base_left`.

Reply:
I might have better naming ideas after I write everything down on a piece of paper.

Reply:
Done.